components/mediation/data-publishers/org.wso2.micro.integrator.prometheus.publisher/src/main/java/org/wso2/micro/integrator/prometheus/publisher/service/MetricResource.java

```diff
@@ -85,4 +99,18 @@ public boolean invoke(MessageContext synCtx) {
         return true;
     }
 
+    private Set<String> parseQuery(String query) throws IOException {
+
+        Set<String> names = new HashSet<>();
+        if (query != null) {
+            String[] pairs = query.split("&");
+            for (String pair : pairs) {
+                int idx = pair.indexOf("=");
+                if (idx != -1 && URLDecoder.decode(pair.substring(0, idx), "UTF-8").equals("name[]")) {
```

Comment:
Do we parse a query parameter called a "name[]"? Does it conform Rest API standards? What's the value of it?